### ISSUE TITLE
[backend] <Sample Pipeline not consuming input IAM role parameter>

### ISSUE THREAD

**Issue by jsitu777** [kubeflow/pipelines#9158]
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)?
EKS 1.25 Kustomize
*  KFP version: 
v1.7.0

*  KFP SDK version: 
v1

### Steps to reproduce
configure example pipeline per https://github.com/kubeflow/pipelines/tree/master/samples/contrib/aws-samples/simple_train_pipeline

Replace these parameters with aws resources created:
https://github.com/kubeflow/pipelines/blob/master/samples/contrib/aws-samples/simple_train_pipeline/training-pipeline.py#L28
https://github.com/kubeflow/pipelines/blob/master/samples/contrib/aws-samples/simple_train_pipeline/training-pipeline.py#L41
https://github.com/kubeflow/pipelines/blob/master/samples/contrib/aws-samples/simple_train_pipeline/training-pipeline.py#L57
https://github.com/kubeflow/pipelines/blob/master/samples/contrib/aws-samples/simple_train_pipeline/training-pipeline.py#L64

### Expected result

Pipeline can be run and complete.

### Error
See this in the pipeline UI log
```
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (AccessDenied) when calling the AssumeRoleWithWebIdentity operation: Not authorized to perform sts:AssumeRoleWithWebIdentity
time="2023-04-14T21:40:33.669Z" level=error msg="cannot save artifact /tmp/outputs/job_name/data" argo=true error="stat /tmp/outputs/job_name/data: no such file or directory"
time="2023-04-14T21:40:33.669Z" level=error msg="cannot save artifact /tmp/outputs/model_artifact_url/data" argo=true error="stat /tmp/outputs/model_artifact_url/data: no such file or directory"
time="2023-04-14T21:40:33.669Z" level=error msg="cannot save artifact /tmp/outputs/training_image/data" argo=true error="stat /tmp/outputs/training_image/data: no such file or directory"
Error: exit status 1
```

Basically the pipeline didn't seem to use the role I specified which has the required S3 and Sagemaker Full Access Permission
